File: Python/Problem_2706.py
from typing import List

class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        min, second_min = 101, 101

        for price in prices:
            if price < min:
                second_min = min
                min = price
            elif price < second_min:
                second_min = price

        return (money - min - second_min) if (money - min - second_min >= 0) else money


if __name__=="__main__":
    sol = Solution()

    print(sol.buyChoco([1,2,2], 3))
    print(sol.buyChoco([3,2,3], 3))

File: Cpp/Problem_2706.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
	int buyChoco(vector<int>& prices, int money) {
		int min = INT_MAX, second_min = INT_MAX;

		for (int price : prices) {
			if (price < min) {
				second_min = min;
				min = price;
			}
			else if (price < second_min)
				second_min = price;
		}

		return min + second_min <= money ? money - min - second_min : money;
	}
};

int main() {
	Solution sol;

	vector<int> input = { 1,2,2 };
	cout << sol.buyChoco(input, 3) << endl;

	input = { 3, 2, 3 };
	cout << sol.buyChoco(input, 3) << endl;

	return 0;
}